src/canmatrix/tests/test_canmatrix.py

```diff
@@ -11,3 +12,22 @@ def test_signal_defaults_to_decimal():
 
     assert isinstance(signal.offset, decimal.Decimal)
     assert isinstance(signal.factor, decimal.Decimal)
+
+
+# SignalGroup
+def test_sinalgroup_add_remove():
```

Comment:
Missing the 'g' in `sinal`.

I'd have a separate test for each feature.  I see add signal adds, add signal doesn't duplicate, group is iterable, `.byName()` finds something, get item access, `.byName()` finds nothing, deleting existent works, deleting not existent doesn't raise, invalid get item access raises.  Separating them out gives more resolution when looking at the failure reports rather than having to read the actual exception and only knowing the first failed test.

I do see that I cheated and checked two things in my sample test.  :|  I'm not a stickler for exactly one assert per test, but I'd tend away from one test for a whole class at least.